tests/cpp/multidevice.h

```diff
@@ -51,4 +51,22 @@ class MultiDeviceTest : public NVFuserTest {
   bool disable_skip;
 };
 
+// This macro is supposed to be used in a test case of a MultiDeviceTest or its
+// `SetUp` method, which have access to GTEST_SKIP and communicator_. It's not
+// made a function because that function wouldn't be able to skip the test by
+// calling GTEST_SKIP.
+#define SKIP_IF_NOT_ENOUGH_DEVICES(fusion)                          \
```

Comment:
I initially moved it so test_multidevice_sharding.cpp could use it too. It's no longer needed as I simplified the testing configs. So I can either move it back or leave it here as a common utility especially for testing non-full meshes.